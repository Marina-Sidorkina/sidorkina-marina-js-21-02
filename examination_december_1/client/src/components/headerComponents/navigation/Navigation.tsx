import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import styles from './Navigation.module.scss';
import { ThemeContext } from '../../../contexts/ThemeContext';
import NavigationPostsIcon from '../navigationPostsIcon/NavigationPostsIcon';
import NavigationUsersIcon from '../navigationUsersIcon/NavigationUsersIcon';
import '../../../locale/i18next';

const Navigation = () => {
  const themeContext = useContext(ThemeContext);
  const { t } = useTranslation();

  return (
    <nav className={styles.navigation}>
      <ul className={styles.list}>
        <li className={styles.item}>
          <Link className={styles.link} to="/users">
            <NavigationUsersIcon
              dark={themeContext.darkTheme}
            />
            <span className={themeContext.darkTheme
              ? `${styles.text} ${styles.text_dark}`
              : styles.text}
            >
              { t('navigation.users', {}) }
            </span>
          </Link>
        </li>
        <li className={styles.item}>
          <Link className={styles.link} to="/posts">
            <NavigationPostsIcon
              dark={themeContext.darkTheme}
            />
            <span className={themeContext.darkTheme
              ? `${styles.text} ${styles.text_dark}`
              : styles.text}
            >
              { t('navigation.posts') }
            </span>
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
