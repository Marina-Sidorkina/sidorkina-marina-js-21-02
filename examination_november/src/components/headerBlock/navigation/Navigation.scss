.navigation {
  width: 240px;
  height: 60px;

  @media (max-width: 769px) {
    width: 130px;
    margin-left: -40px;
  }

  @media (max-width: 481px) {
    width: 80px;
    margin-left: -15px;
  }

  &__list {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 240px;
    height: 60px;
    padding: 0;
    margin: 0;
    list-style: none;
    font-size: 16px;

    @media (max-width: 769px) {
      width: 130px;
    }

    @media (max-width: 481px) {
      width: 80px;
    }
  }

  &__item {
    box-sizing: border-box;
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;

    &:hover {
      opacity: 0.7;
    }
  }

  &__svg {
    margin-right: 7px;

    &_users {
      width: 24px;
      height: 24px;
      stroke: #000000;

      &_dark {
        stroke: #c9d1d9;
      }
    }

    &_posts {
      width: 24px;
      height: 24px;
      fill: #000000;

      &_dark {
        fill: #c9d1d9;
      }
    }
  }

  &__text {

    &_dark {
      color: #c9d1d9;
    }

    @media (max-width: 769px) {
      display: none;
    }
  }

  &__link {
    display: flex;
    align-items: center;
    color: #000000;

    &:hover {
      color: #000000;
    }
  }
}
