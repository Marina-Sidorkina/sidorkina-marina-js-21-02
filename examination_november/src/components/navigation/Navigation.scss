.navigation {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 240px;
  height: 60px;
  padding: 0;
  margin: 0;
  list-style: none;
  font-size: 16px;

  &__item {
    box-sizing: border-box;
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;

    &:hover {
      opacity: 0.7;
    }
  }

  &__svg {
    margin-right: 7px;

    &_users {
      width: 24px;
      height: 24px;
      stroke: #000000;
    }

    &_posts {
      width: 24px;
      height: 24px;
      fill: #000000;
    }
  }
}
