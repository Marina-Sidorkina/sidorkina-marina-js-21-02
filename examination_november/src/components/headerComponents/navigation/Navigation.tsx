import React, { useContext } from 'react';
import './Navigation.scss';
import { Link } from 'react-router-dom';
import { ThemeContext } from '../../../contexts/ThemeContext';
import NavigationPostsIcon from '../navigationPostsIcon/NavigationPostsIcon';
import NavigationUsersIcon from '../navigationUsersIcon/NavigationUsersIcon';

const Navigation = () => {
  const themeContext = useContext(ThemeContext);

  return (
    <nav className="navigation">
      <ul className="navigation__list">
        <li className="navigation__item navigation__item_users">
          <Link className="navigation__link" to="/users">
            <NavigationUsersIcon
              dark={themeContext.darkTheme}
            />
            <span className={`${themeContext.darkTheme
              ? 'navigation__text navigation__text_dark'
              : 'navigation__text'}`}
            >
              Пользователи
            </span>
          </Link>
        </li>
        <li className="navigation__item navigation__item_posts">
          <Link className="navigation__link" to="/posts">
            <NavigationPostsIcon
              dark={themeContext.darkTheme}
            />
            <span className={`${themeContext.darkTheme
              ? 'navigation__text navigation__text_dark'
              : 'navigation__text'}`}
            >
              Посты
            </span>
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
