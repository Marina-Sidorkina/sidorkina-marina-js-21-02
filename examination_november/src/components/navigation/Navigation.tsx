import React from 'react';
import './Navigation.scss';

const Navigation = () => (
  <nav className="navigation">
    <ul className="navigation__list">
      <li className="navigation__item navigation__item_users">
        <svg
          className="navigation__svg navigation__svg_users"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          fill="none"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
        >
          <path d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2" />
          <circle cx="9" cy="7" r="4" />
          <path d="M23 21v-2a4 4 0 0 0-3-3.87" />
          <path d="M16 3.13a4 4 0 0 1 0 7.75" />
        </svg>
        <span>Пользователи</span>
      </li>
      <li className="navigation__item navigation__item_posts">
        <svg
          className="navigation__svg navigation__svg_posts"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
          height="30"
          viewBox="0 0 96 96"
        >
          <path
            d="M84 24v48c0 3.312-2.688 6-6 6h-63c0 3.312 2.688 6 6
          6h63c3.312 0 6-2.688 6-6v-48c0-3.312-2.688-6-6-6zM81
          69v-48c0-3.312-2.688-6-6-6h-63c-3.312 0-6 2.688-6 6v48c0
          3.312 2.688 6 6 6h63c3.312 0 6-2.688 6-6zM12 21h63v48h-63v-48zM46.881
          51.933l-8.964-9.543-8.967 15.312-6.726-7.029-7.287 15.387h57.165l-12.891-33.735-12.33
          19.608zM24 42c3.312 0 6-2.688 6-6s-2.688-6-6-6-6 2.688-6 6 2.688 6 6 6z"
          />
        </svg>
        <span>Посты</span>
      </li>
    </ul>
  </nav>
);

export default Navigation;
