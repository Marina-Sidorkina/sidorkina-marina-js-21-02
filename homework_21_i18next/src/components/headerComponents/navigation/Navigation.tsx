import React, { useContext } from 'react';
import './Navigation.scss';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { ThemeContext } from '../../../contexts/ThemeContext';
import NavigationPostsIcon from '../navigationPostsIcon/NavigationPostsIcon';
import NavigationUsersIcon from '../navigationUsersIcon/NavigationUsersIcon';
import '../../../locale/i18next';

const Navigation = () => {
  const themeContext = useContext(ThemeContext);
  const { t } = useTranslation();

  return (
    <nav className="navigation">
      <ul className="navigation__list">
        <li className="navigation__item navigation__item_users">
          <Link className="navigation__link" to="/users">
            <NavigationUsersIcon
              dark={themeContext.darkTheme}
            />
            <span className={`${themeContext.darkTheme
              ? 'navigation__text navigation__text_dark'
              : 'navigation__text'}`}
            >
              { t('navigation.users', {}) }
            </span>
          </Link>
        </li>
        <li className="navigation__item navigation__item_posts">
          <Link className="navigation__link" to="/posts">
            <NavigationPostsIcon
              dark={themeContext.darkTheme}
            />
            <span className={`${themeContext.darkTheme
              ? 'navigation__text navigation__text_dark'
              : 'navigation__text'}`}
            >
              { t('navigation.posts') }
            </span>
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
