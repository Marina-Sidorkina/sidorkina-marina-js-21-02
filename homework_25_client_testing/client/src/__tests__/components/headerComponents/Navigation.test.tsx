import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { HashRouter } from 'react-router-dom';
import Navigation from '../../../components/headerComponents/navigation/Navigation';

configure({ adapter: new Adapter() });

describe('Navigation component test', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(<HashRouter><Navigation /></HashRouter>);
  });

  test('should render Navigation', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('should render Navigation', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
