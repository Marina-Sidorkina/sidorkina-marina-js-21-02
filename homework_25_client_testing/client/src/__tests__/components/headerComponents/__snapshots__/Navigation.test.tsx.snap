// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component test should render Navigation 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navigation>
      <nav
        className="navigation"
      >
        <ul
          className="list"
        >
          <li
            className="item"
          >
            <Link
              className="link"
              to="/users"
            >
              <LinkAnchor
                className="link"
                href="#/users"
                navigate={[Function]}
              >
                <a
                  className="link"
                  href="#/users"
                  onClick={[Function]}
                >
                  <NavigationUsersIcon>
                    <svg
                      className="svg svg_users"
                      fill="none"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
                      />
                      <circle
                        cx="9"
                        cy="7"
                        r="4"
                      />
                      <path
                        d="M23 21v-2a4 4 0 0 0-3-3.87"
                      />
                      <path
                        d="M16 3.13a4 4 0 0 1 0 7.75"
                      />
                    </svg>
                  </NavigationUsersIcon>
                  <span
                    className="text"
                  >
                    Users
                  </span>
                </a>
              </LinkAnchor>
            </Link>
          </li>
          <li
            className="item"
          >
            <Link
              className="link"
              to="/posts"
            >
              <LinkAnchor
                className="link"
                href="#/posts"
                navigate={[Function]}
              >
                <a
                  className="link"
                  href="#/posts"
                  onClick={[Function]}
                >
                  <NavigationPostsIcon>
                    <svg
                      className="svg svg_posts"
                      height="30"
                      viewBox="0 0 96 96"
                      width="30"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M84 24v48c0 3.312-2.688 6-6 6h-63c0 3.312 2.688 6 6 6h63c3.312 0 6-2.688 6-6v-48c0-3.312-2.688-6-6-6zM81 69v-48c0-3.312-2.688-6-6-6h-63c-3.312 0-6 2.688-6 6v48c0 3.312 2.688 6 6 6h63c3.312 0 6-2.688 6-6zM12 21h63v48h-63v-48zM46.881 51.933l-8.964-9.543-8.967 15.312-6.726-7.029-7.287 15.387h57.165l-12.891-33.735-12.33 19.608zM24 42c3.312 0 6-2.688 6-6s-2.688-6-6-6-6 2.688-6 6 2.688 6 6 6z"
                      />
                    </svg>
                  </NavigationPostsIcon>
                  <span
                    className="text"
                  >
                    Posts
                  </span>
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </ul>
      </nav>
    </Navigation>
  </Router>
</HashRouter>
`;
